examples/DEVELOPER.md

```diff
@@ -208,3 +208,59 @@ export UPARGS="--scale http_service=2"
 If your example asks the user to run commands inside containers, you can
 mimick this using `docker-compose exec -T`. The `-T` flag is necessary as the
 tests do not have access to a `tty` in the CI pipeline.
+
+## Add sandbox configs to configuration tests
+
+Example configuration files are tested to ensure they are valid and well-formed, and do
+not contain deprecated features.
+
+### Exclude configs from example configuration tests
+
+The CI script to find the configuration files will search for all files with a `yaml` or `lua`
+extension. If your sandbox example contains non-envoy configuration files that should not be
+checked you can exclude them by setting the `EXCLUDED_BUILD_CONFIGS` in `ci/verify_examples.sh`.
+
+Given a sandbox with the name `example-sandbox`, and a configuration file that should be excluded
+named `my-config.yaml`, add `^./example-sandbox/my-config.yaml` to `EXCLUDED_BUILD_CONFIGS`
+
+For example, change:
+```
+EXCLUDED_BUILD_CONFIGS=${EXCLUDED_BUILD_CONFIGS:-"^./cache/responses.yaml|^./jaeger-native-tracing|docker-compose"}
+```
+
+to:
+```
+EXCLUDED_BUILD_CONFIGS=${EXCLUDED_BUILD_CONFIGS:-"^./example-sandbox/my-config.yaml|^./cache/responses.yaml|^./jaeger-native-tracing|docker-compose"}
+```
+
+This variable is used as a regular expression, so you can exclude multiple or more targeted exclusions using an
+appropriate bash `regex`.
+
+### List configs in `BUILD` file
+
+Configuration files that have been added to your sandbox should be listed in ``examples/BUILD``, in
+the ``filegroup.srcs`` section.
+
+### Increment the expected number of configuration files in `example_configs_test.cc`
+
+You will also need to increment the number of expected files in ``test/config_test/example_configs_test.cc``
+
+Find the section shown in the following code block:
+
+```c++
+#if defined(__APPLE__) || defined(WIN32)
+  // freebind/freebind.yaml is not supported on macOS or Windows and is disabled via Bazel.
+  EXPECT_EQ(37UL, ConfigTest::run(directory));
```

Comment:
Im wondering if its possible to shift this int to a constant - its not immediately obvious to me how best to do this

Reply:
Yeah, probably you can do the following?

```diff
diff --git a/test/config_test/example_configs_test.cc b/test/config_test/example_configs_test.cc
index 2e02d41608..2ccfa9a41c 100644
--- a/test/config_test/example_configs_test.cc
+++ b/test/config_test/example_configs_test.cc
@@ -5,6 +5,10 @@
 #include "gtest/gtest.h"
 
 namespace Envoy {
+
+// The total number of available configuration files under the examples directory.
+constexpr uint64_t NumberOfConfigurationFiles = 38UL;
+
 TEST(ExampleConfigsTest, All) {
   TestEnvironment::exec(
       {TestEnvironment::runfilesPath("test/config_test/example_configs_test_setup.sh")});
@@ -21,9 +25,9 @@ TEST(ExampleConfigsTest, All) {
 
 #if defined(__APPLE__) || defined(WIN32)
   // freebind/freebind.yaml is not supported on macOS or Windows and is disabled via Bazel.
-  EXPECT_EQ(37UL, ConfigTest::run(directory));
+  EXPECT_EQ(NumberOfConfigurationFiles - 1, ConfigTest::run(directory));
 #else
-  EXPECT_EQ(38UL, ConfigTest::run(directory));
+  EXPECT_EQ(NumberOfConfigurationFiles, ConfigTest::run(directory));
 #endif
```

Reply:
Not sure if you want to go further by getting that `NumberOfConfigurationFiles` from e.g. env var.

Reply:
yep, or even the actual count of config files

Reply:
ill update with your suggestion on how to move to constant and we can figure it out from there

Reply:
Yeah, I mean you can possibly "count" the number of config files in `example_configs_test_setup.sh` and set an env var and consume that from the test.

Reply:
yep. i think ill separate to a different PR. Ill create it later today